include/z64save.h

```diff
@@ -252,6 +252,16 @@ typedef enum {
     /* 3 */ GAMEMODE_END_CREDITS
 } GameMode;
 
+typedef enum {
+    /* 0 */ SCENE_LAYER_CHILD_DAY,
+    /* 1 */ SCENE_LAYER_CHILD_NIGHT,
+    /* 2 */ SCENE_LAYER_ADULT_DAY,
+    /* 3 */ SCENE_LAYER_ADULT_NIGHT    
+} SceneLayer;
+
+#define SCENE_LAYER_CUTSCENE_START (SCENE_LAYER_ADULT_NIGHT + 1)
+#define IS_CUTSCENE_LAYER (gSaveContext.sceneLayer >= SCENE_LAYER_CUTSCENE_START)
```

Comment:
I feel like `IS_CUTSCENE_LAYER(gSaveContext.sceneLayer)` (pass the layer as argument) would be more readable, and there are conditional statements where `gSaveContext.sceneLayer` is explicit anyway, so not having it in some looks inconsistent?

Which also makes me wonder if a macro is worth it at all

Reply:
i think it makes sense for it to be explicit when its checking for a specific layer, and not when its not. It is actually pretty helpful to search `gSaveContext.sceneLayer` and see the special cases that arent generalized to "is or is not a cutscene"

dont see a purpose in the arg being supplied personally. its like IS_DAY or IS_NIGHT which is making a more general statement and abstracting the variable used to determine that

Reply:
to elaborate, I see the macro as a general query of "is the current scene layer a cutscene?"
not
"is this layer a cutscene"

Reply:
but with `IS_DAY` you only have a few usage of `gSaveContext.nightFlag` (for setting it and indexing things), while `gSaveContext.sceneLayer` and the macro hiding it are used in the same context

Reply:
the context is different tho. when you see the variable accessed directly, it concerns code about a very specific layer. when this macro is used it is a generalization about them